File: 1.0/src/c/cz/mg/vulkan/PFN_vkDestroySamplerYcbcrConversionKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroySamplerYcbcrConversionKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong ycbcrConversion, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroySamplerYcbcrConversionKHR f = (PFN_vkDestroySamplerYcbcrConversionKHR)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkSamplerYcbcrConversion*)jniLongToPointer(ycbcrConversion)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkDestroySamplerYcbcrConversionKHR.java
package cz.mg.vulkan;

public class PFNvkDestroySamplerYcbcrConversionKHR extends VkFunctionPointer {
    public PFNvkDestroySamplerYcbcrConversionKHR() {
    }

    public PFNvkDestroySamplerYcbcrConversionKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkDestroySamplerYcbcrConversionKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroySamplerYcbcrConversionKHR(long value) {
        setValue(value);
    }

    public PFNvkDestroySamplerYcbcrConversionKHR(VkInstance instance) {
        super(instance, new VkString("vkDestroySamplerYcbcrConversionKHR"));
    }

    public void call(VkDevice device, VkSamplerYcbcrConversionKHR ycbcrConversion, VkAllocationCallbacks pAllocator){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), ycbcrConversion != null ? ycbcrConversion.getVkAddress() : VkPointer.getNullAddress(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void call(long vkaddress, long device, long ycbcrConversion, long pAllocator);
}
